### PR TITLE
AddressBar trigger implemented. Fixed warnings on non existing contex…

### PR DESCRIPTION
**Pull request by vladest**
…t properties. Fixed warnings about non notifyable property

https://highfidelity.manuscript.com/f/cases/9223/The-Enter-key-does-not-close-the-address-bar-in-hmd

Test plan:
- Open address bar pressing enter
- Move focus out of address input field by clicking on empty space within Tablet
- Press enter. Tablet should close

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/6157/app_12126_PR.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/6157/HighFidelity-Beta-PR12126-075bcba49f2fb88b542e2d5e688759a89093231c.dmg

**Comment by sethalves** [comment]
build this please

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/6165/app_12126_PR.apk

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/6165/HighFidelity-Beta-PR12126-075bcba49f2fb88b542e2d5e688759a89093231c.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/6165/HighFidelity-Beta-PR12126-075bcba49f2fb88b542e2d5e688759a89093231c.dmg

**Comment by sbennett77** [comment]
Currently Testing.

**Comment by sbennett77** [comment]
Currently Testing.

**Comment by sbennett77** [comment]
Passed.

@conklin94122

File: interface/src/ui/overlays/Web3DOverlay.cpp [hunk 1]
```diff
@@ -72,6 +72,13 @@ Web3DOverlay::Web3DOverlay() {
     connect(this, &Web3DOverlay::requestWebSurface, this, &Web3DOverlay::buildWebSurface);
     connect(this, &Web3DOverlay::releaseWebSurface, this, &Web3DOverlay::destroyWebSurface);
     connect(this, &Web3DOverlay::resizeWebSurface, this, &Web3DOverlay::onResizeWebSurface);
+
+    //need to be intialized before Tablet 1st open
+    _webSurface = DependencyManager::get<OffscreenQmlSurfaceCache>()->acquire(_url);
```
Comment (zfox23) [inline interface/src/ui/overlays/Web3DOverlay.cpp]: Can you please help me understand what this is doing differently from the code in `Application.cpp` inside `initializeUi()` such as `surfaceContext->setContextProperty("Account", GlobalServicesScriptingInterface::getInstance());`?

Comment (vladest) [reply interface/src/ui/overlays/Web3DOverlay.cpp]: I suppose, scope is different: one context is visible by scripts only, another one is for Tablet UI

**Comment by sethalves** [comment]
When a text field gets focus while in HMD, the virtual keybord should open (if not already open).
After a text field looses focus while in HMD, the virtual keyboard should close (if focus is not being changed to a new text field).

Virtual keyboard should have option to close. (A button.) Closing it should not loose focus.
Virtual keyboard should show underscore on shift.


A mixed input app such as the address bar, in which the text input and other inputs are both primary, should give focus to the text input when it is opened via the keyboard regardless of whether in HMD. (In HMD, this will also bring up the virtual keyboard.) Other means of opening (such as menu, or button press,) do not focus on the text.


A "click" inside (or outside) of a text box should gain (or loose) focus:

    laser/stylus/mouse on qml
    laser/stylus/mouse on html
    laser/mouse outside of the tablet or hud.

File: interface/src/ui/overlays/Web3DOverlay.cpp (hunk 1 above)
Comment (jherico) [reply interface/src/ui/overlays/Web3DOverlay.cpp]: @vladest `surfaceContext->setContextProperty` in Application.cpp doesn't make anything visible to scripts.  It makes things visible to the QML objects loaded onto the main HUD.  Exposure to scripts is currently handled by `scriptEngine->registerGlobalObject`

**Pull request merged**